gluten-core/src/main/scala/org/apache/gluten/GlutenPlugin.scala

```diff
@@ -134,7 +134,11 @@ private[gluten] class GlutenDriverPlugin extends DriverPlugin with Logging {
     conf.set(SPARK_SESSION_EXTENSIONS.key, extensions)
 
     // adaptive custom cost evaluator class
-    if (GlutenConfig.getConf.enableGluten && GlutenConfig.getConf.enableGlutenCostEvaluator) {
+    val glutenEnabled = conf.getBoolean(GlutenConfig.GLUTEN_ENABLED_KEY, GLUTEN_ENABLED_BY_DEFAULT)
+    lazy val enableGlutenCostEvaluator = conf.getBoolean(
+      GlutenConfig.GLUTEN_COST_EVALUATOR_ENABLED,
+      GLUTEN_COST_EVALUATOR_DEFAULT_VALUE)
+    if (glutenEnabled && enableGlutenCostEvaluator) {
       val costEvaluator = "org.apache.spark.sql.execution.adaptive.GlutenCostEvaluator"
       conf.set(SQLConf.ADAPTIVE_CUSTOM_COST_EVALUATOR_CLASS.key, costEvaluator)
     }
```

Comment:
It turns out that this part of code is problematical.

The entrypoint `GlutenPlugin` is called only once in the same Spark session. So say user could set `spark.gluten.enabled=false` then set it to true using `SET` prompt after a while.

Will there be any issue if we just make `GlutenCostEvaluator` always load? Which means removing the conditions, namely `if (glutenEnabled && enableGlutenCostEvaluator)` here.

cc @zml1206

Reply:
There seems to be no problem with always loading GlutenCostEvaluator, because GlutenCostEvaluator will determine whether gluten is enabled.

Reply:
should we only retain the `enableGlutenCostEvaluator`  condition whether to enable gluten?  @zml1206 @zhztheplayer

Reply:
> Should we only retain the `enableGlutenCostEvaluator` condition, no matter whether to enable gluten? @zml1206 @zhztheplayer

Yes, let's try remove the if conditions.